velox/vector/fuzzer/VectorFuzzer.cpp

```diff
@@ -865,6 +867,69 @@ RowVectorPtr VectorFuzzer::fuzzRowChildrenToLazy(RowVectorPtr rowVector) {
       std::move(children));
 }
 
+// Utility function to check if a RowVector can have nested lazy children.
+bool canWrapChildrenInLazy(const VectorPtr& baseVector) {
+  if (baseVector->typeKind() == TypeKind::ROW) {
+    if (VectorEncoding::isDictionary(baseVector->encoding())) {
+      return canWrapChildrenInLazy(baseVector->valueVector());
+    }
+    RowVector* rowVector = baseVector->as<RowVector>();
+    if (rowVector) {
+      return rowVector->nulls() == nullptr;
+    }
+  }
+
+  return false;
+}
+
+// Utility function That only takes a row vector that passes the check in
+// canWrapChildrenInLazy() and either picks the first child to be wrapped in
+// lazy OR picks the first row vector child that passes canWrapChildrenInLazy()
+// and traverses its children recursively.
+VectorPtr wrapInnermostChildToLazy(VectorPtr& baseVector) {
```

Comment:
nit: What about "wrapRowChildInLazyRecursive"? Because this function doesn't just wrap the innermost child.